frontend/src/modules/machinery/components/Machinery.tsx

```diff
@@ -26,142 +24,125 @@ type State = {
  * strings, coming from various sources like Translation Memory or
  * third-party Machine Translation.
  */
-export class Machinery extends React.Component<Props, State> {
-    searchInput: { current: any };
-
-    constructor(props: Props) {
-        super(props);
-        this.searchInput = React.createRef();
-        this.state = { page: 1 };
-    }
-
-    componentDidMount() {
-        const { machinery } = this.props;
-
+export const Machinery = ({
+    locale,
+    machinery,
+    searchMachinery,
+}: Props): null | React.ReactElement<'section'> => {
+    const [page, setPage] = React.useState(1);
+    const searchInput = React.useRef<HTMLInputElement>();
+    const prevEntity = usePrevious(machinery.entity);
+    const prevPage = usePrevious(page);
+
+    React.useEffect(() => {
         // Restore custom input in search field,
         // e.g. when switching from Locales tab back to Machinery
-        if (!machinery.entity && machinery.sourceString) {
-            this.searchInput.current.value = machinery.sourceString;
+        if (machinery.searchString) {
+            searchInput.current.value = machinery.searchString;
         }
-    }
-
-    componentDidUpdate(prevProps: Props) {
-        const { machinery } = this.props;
+    }, []);
 
+    React.useEffect(() => {
         // Clear search field after switching to a different entity
-        if (
-            machinery.entity &&
-            machinery.entity !== prevProps.machinery.entity
-        ) {
-            this.searchInput.current.value = '';
-            this.setState({ page: 1 });
+        if (!machinery.searchString && machinery.entity !== prevEntity) {
+            searchInput.current.value = '';
+            setPage(1);
         }
-    }
+    }, [machinery.entity]);
 
-    handleResetSearch: () => void = () => {
-        this.props.searchMachinery('');
-        this.setState({ page: 1 });
+    React.useEffect(() => {
+        // Fetch next page of search results
+        if (prevPage && page !== prevPage) {
+            searchMachinery(searchInput.current.value, page);
+        }
+    }, [page, searchMachinery]);
+
+    const [sentryRef, { rootRef }] = useInfiniteScroll({
+        loading: machinery.fetching,
+        hasNextPage: machinery.hasMore,
+        onLoadMore: () => setPage((page) => page + 1),
+        rootMargin: '0px 0px 400px 0px',
+    });
+
+    const resetSearch = () => {
+        searchMachinery('');
+        searchInput.current.value = '';
+        setPage(1);
     };
 
-    submitForm: (event: React.SyntheticEvent<HTMLFormElement>) => void = (
-        event: React.SyntheticEvent<HTMLFormElement>,
-    ) => {
+    const submitForm = (event: React.SyntheticEvent<HTMLFormElement>) => {
         event.preventDefault();
-        this.props.searchMachinery(this.searchInput.current.value);
+        setPage(1);
```

Comment:
This is needed in order to reset the page count if a user performs an additional search without first changing entities or resetting the search input field.